### PR TITLE
[Do No Merge]test contrib build with fips

### PR DESCRIPTION
**Pull request by soulxu**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: ci/do_ci.sh [hunk 1]
```diff
@@ -463,11 +463,12 @@ case $CI_TARGET in
             --define log_debug_assert_in_release=disabled
         echo "Building binary with wasm=wasmtime... and logging disabled"
         bazel build "${BAZEL_BUILD_OPTIONS[@]}" \
+            --action_env=TEST=TEST \
             --config=compile-time-options \
             --define wasm=wasmtime \
             --define enable_logging=disabled \
             -c fastbuild \
-            @envoy//source/exe:envoy-static \
+            @envoy//contrib/exe:envoy-static \
```
Comment (phlax) [inline ci/do_ci.sh:471]: we kinda want the test with the 2 following conditions to compare:

- existing - ie `//source`
- proposed change - ie _both_

File: ci/do_ci.sh [hunk 2]
```diff
@@ -463,11 +463,12 @@ case $CI_TARGET in
             --define log_debug_assert_in_release=disabled
         echo "Building binary with wasm=wasmtime... and logging disabled"
         bazel build "${BAZEL_BUILD_OPTIONS[@]}" \
+            --action_env=TEST=TEST \
```
Comment (phlax) [inline ci/do_ci.sh:466]: for any timed test runs `TEST=TEST` will need to be changed to eg `TEST=TEST23`

Comment (phlax) [reply ci/do_ci.sh:466]: seems this messed up the ICU build - was wondering whether setting `TEST` would be safe - probs make it something like `ARBITRARY_TEST`

File: ci/do_ci.sh (hunk 1 above)
Comment (soulxu) [reply ci/do_ci.sh:471]: ok, let me try to build those two in c-t-o

File: ci/do_ci.sh (hunk 2 above)
Comment (soulxu) [reply ci/do_ci.sh:466]: emm...yea, TEST may be used by something.

File: ci/do_ci.sh (hunk 1 above)
Comment (phlax) [reply ci/do_ci.sh:480]: cool - was wondering whether we would need to separate as you have done

the benefit of this is that we only need to run the ci once to get the timings, and can see exactly how long the contrib build takes

Comment (soulxu) [reply ci/do_ci.sh:480]: I run two bazel cmds, suppose we can get the timings separately. Let me trigger the CI it seems the CI broken friday.

**Comment by soulxu** [comment]
I don't know why this failure https://dev.azure.com/cncf/envoy/_build/results?buildId=169058&view=logs&j=e969334a-0e55-5c18-ac96-8b546753391e&t=d2cfa5df-b334-5e06-ef04-7a1905627a50&l=114, let me try to build one by one

**Comment by phlax** [comment]
> I don't know why this failure ...

it has increased the cache size, and now its running out of (tmpfs) space to load the cache

**Pull request closed without merging**